docker/Dockerfile

```diff
@@ -63,26 +62,26 @@ RUN export USE_CUDA=1
 
 ARG CUDA_VERSION=""
 
-RUN git clone --depth 1 https://github.com/pytorch/serve.git
-
-WORKDIR "serve"
-
-RUN \
+RUN TORCH_VER=$(curl --silent --location https://pypi.org/pypi/torch/json | python -c "import sys, json, pkg_resources; releases = json.load(sys.stdin)['releases']; print(sorted(releases, key=pkg_resources.parse_version)[-1])") && \
+    TORCH_VISION_VER=$(curl --silent --location https://pypi.org/pypi/torchvision/json | python -c "import sys, json, pkg_resources; releases = json.load(sys.stdin)['releases']; print(sorted(releases, key=pkg_resources.parse_version)[-1])") && \
+    TORCH_TEXT_VER=$(curl --silent --location https://pypi.org/pypi/torchtext/json | python -c "import sys, json, pkg_resources; releases = json.load(sys.stdin)['releases']; print(sorted(releases, key=pkg_resources.parse_version)[-1])") && \
+    TORCH_AUDIO_VER=$(curl --silent --location https://pypi.org/pypi/torchaudio/json | python -c "import sys, json, pkg_resources; releases = json.load(sys.stdin)['releases']; print(sorted(releases, key=pkg_resources.parse_version)[-1])") && \
     if echo "$BASE_IMAGE" | grep -q "cuda:"; then \
         # Install CUDA version specific binary when CUDA version is specified as a build arg
         if [ "$CUDA_VERSION" ]; then \
-            python ./ts_scripts/install_dependencies.py --cuda $CUDA_VERSION; \
-        # Install the binary with the latest CPU image on a CUDA base image
+            python -m pip install --no-cache-dir torch==$TORCH_VER+$CUDA_VERSION torchvision==$TORCH_VISION_VER+$CUDA_VERSION torchtext==$TORCH_TEXT_VER torchaudio==$TORCH_AUDIO_VER+$CUDA_VERSION -f https://download.pytorch.org/whl/torch_stable.html; \
+        # Install CPU binaries on a CUDA enabled device
         else \
-            python ./ts_scripts/install_dependencies.py; \
+            python -m pip install --no-cache-dir torch==$TORCH_VER+cpu torchvision==$TORCH_VISION_VER+cpu torchtext==$TORCH_TEXT_VER+cpu torchaudio==$TORCH_AUDIO_VER+cpu -f https://download.pytorch.org/whl/torch_stable.html; \
```

Comment:
I see that the new behavior is equivalent with previous one but just out of curiosity: Why do we install the cpu version in a cuda base image and don't fall back to a default cuda version (like the newest)? Is there a use case for cpu torch in cuda base image?

Reply:
Yes, thats right. CPU binaries on a CUDA enabled device

Reply:
Thanks, good point. Let me check and correct it. I am not sure of the historical reason why we had this use-case. @msaroufim ?

Reply:
What if someone has more than one model they're running in a container and they want to run one on GPU and another on CPU

Reply:
Maybe I am missing your point but in that case they still need to use the gpu docker image as they will need torch-gpu for the model running on GPU. The line creates an nvidia docker with torch-cpu installed, so you could only run a CPU model in such an image.

Reply:
Oh i see now, yes torch cpu is not useful on an nvidia docker image